frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/hooks.ts

```diff
@@ -262,3 +262,29 @@ export const useNodes = (
 
   return [...nodes, finallyNode];
 };
+
+const touchTaskFields = (task: PipelineTask): FormikTouched<PipelineTask> => ({
+  workspaces: task.workspaces?.map(() => ({ workspace: true })),
+  resources: {
+    inputs: task.resources?.inputs?.map(() => ({ resource: true })),
+    outputs: task.resources?.outputs?.map(() => ({ resource: true })),
+  },
+});
+
+export const useExplicitPipelineTaskTouch = () => {
+  const { setTouched, touched, values } = useFormikContext<PipelineBuilderFormikValues>();
+  const workspacesTouched = !!touched.formData?.workspaces;
+  const resourcesTouched = !!touched.formData?.resources;
+
+  React.useEffect(() => {
+    if (workspacesTouched || resourcesTouched) {
+      setTouched({
+        formData: {
+          tasks: values.formData?.tasks?.map(touchTaskFields),
+          finallyTasks: values.formData?.finallyTasks?.map(touchTaskFields),
+        },
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
First of all: I tested it locally and it works fine to get an error message when removing a used (untouched) workspace :+1:

Reducing code is welcome in most cases. But in this case I don't see a reason to touch all resources when a workspace is touched and vice versa. Wdyt?

Reply:
I'm also fine with this change if Andrew approve it. It fixes the issue and have no "big disadvantage" :)

Reply:
Thanks, updated to only touch task workspaces if workspaces list is changed, same for resources.